drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,735 @@
+
+            sx127x_reg_write(dev, SX127X_REG_DIOMAPPING1, 0x00);
+            break;
+        default:
+            break;
+    }
+}
+
+uint8_t sx127x_get_syncword(sx127x_t *dev)
+{
+    return sx127x_reg_read(dev, SX127X_REG_LR_SYNCWORD);
+}
+
+void sx127x_set_syncword(sx127x_t *dev, uint8_t syncword)
+{
+    sx127x_reg_write(dev, SX127X_REG_LR_SYNCWORD, syncword);
+}
+
+uint32_t sx127x_get_channel(sx127x_t *dev)
+{
+    return (((uint32_t)sx127x_reg_read(dev, SX127X_REG_FRFMSB) << 16) |
+            (sx127x_reg_read(dev, SX127X_REG_FRFMID) << 8) |
+            (sx127x_reg_read(dev, SX127X_REG_FRFLSB))) * SX127X_FREQUENCY_RESOLUTION;
+}
+
+void sx127x_set_channel(sx127x_t *dev, uint32_t freq)
+{
+    /* Save current operating mode */
+    dev->settings.channel = freq;
+    uint8_t prev_mode = sx127x_reg_read(dev, SX127X_REG_OPMODE);
+
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_STANDBY);
+
+    freq = (uint32_t)((double) freq / (double) SX127X_FREQUENCY_RESOLUTION);
+
+    /* Write frequency settings into chip */
+    sx127x_reg_write(dev, SX127X_REG_FRFMSB, (uint8_t)((freq >> 16) & 0xFF));
+    sx127x_reg_write(dev, SX127X_REG_FRFMID, (uint8_t)((freq >> 8) & 0xFF));
+    sx127x_reg_write(dev, SX127X_REG_FRFLSB, (uint8_t)(freq & 0xFF));
+
+    /* Restore previous operating mode */
+    sx127x_reg_write(dev, SX127X_REG_OPMODE, prev_mode);
+}
+
+uint32_t sx127x_get_time_on_air(sx127x_t *dev)
+{
+    uint32_t air_time = 0;
+
+    uint8_t pkt_len = dev->settings.time_on_air_pkt_len;
+    sx127x_radio_modems_t modem = dev->settings.modem;
+    switch (modem) {
+        case SX127X_MODEM_FSK:
+            break;
+
+        case SX127X_MODEM_LORA:
+        {
+            double bw = 0.0;
+
+            /* Note: When using LoRa modem only bandwidths 125, 250 and 500 kHz are supported. */
+            switch (dev->settings.lora.bandwidth) {
+                case SX127X_BW_125_KHZ:
+                    bw = 125e3;
+                    break;
+                case SX127X_BW_250_KHZ:
+                    bw = 250e3;
+                    break;
+                case SX127X_BW_500_KHZ:
+                    bw = 500e3;
+                    break;
+                default:
+                    DEBUG("Invalid bandwith: %d\n", dev->settings.lora.bandwidth);
+                    break;
+            }
+
+            /* Symbol rate : time for one symbol [secs] */
+            double rs = bw / (1 << dev->settings.lora.sf);
+            double ts = 1 / rs;
+
+            /* time of preamble */
+            double t_preamble = (dev->settings.lora.preamble_len + 4.25) * ts;
+
+            /* Symbol length of payload and time */
+            double tmp =
+                ceil(
+                    (8 * pkt_len - 4 * dev->settings.lora.sf + 28
+                     + 16 * dev->settings.lora.crc_on
+                     - (!dev->settings.lora.implicit_header_mode ? 20 : 0))
+                    / (double) (4 * dev->settings.lora.sf
```

Comment:
I could only find one (compared with [SX1276](https://github.com/Lora-net/LoRaMac-node/blob/master/src/radio/sx1276/sx1276.c) and [SX1272](https://github.com/Lora-net/LoRaMac-node/blob/master/src/radio/sx1272/sx1272.c))